src/transport/CryptoContext.cpp

```diff
@@ -50,39 +51,34 @@ constexpr uint8_t RSEKeysInfo[] = { 0x53, 0x65, 0x73, 0x73, 0x69, 0x6f, 0x6e, 0x
 
 using namespace Crypto;
 
-#ifdef ENABLE_HSM_HKDF
-using HKDF_sha_crypto = HKDF_shaHSM;
-#else
-using HKDF_sha_crypto = HKDF_sha;
-#endif
-
 CryptoContext::CryptoContext() : mKeyAvailable(false) {}
 
 CryptoContext::~CryptoContext()
 {
-    for (auto & key : mKeys)
+    if (mKeystore)
     {
-        ClearSecretData(key, sizeof(CryptoKey));
+        mKeystore->DestroyKey(mEncryptionKey);
+        mKeystore->DestroyKey(mDecryptionKey);
     }
+
+    ClearSecretData(mAttestationChallenge, sizeof(mAttestationChallenge));
     mKeyContext = nullptr;
```

Comment:
Suggest set mKeystore to nullptr for symmetry (even though it should not be needed), as it would possibly help in case of double free.

Reply:
Done